File: bgpateng/push.py
import os
from inery.Api import Api
from inery.JsonRpc import JsonRpc
from inery.JsSignatureProvider import JsSignatureProvider

url = "http://85.239.232.206:8888"

json_rpc = JsonRpc(url)
private_key = os.getenv("PRIVATE_KEY")

account = "bgpateng"
actor = "bgpateng"
signature = JsSignatureProvider([private_key])

api = Api(rpc=json_rpc, signatureProvider=signature)

async def CreateTransaction(id, user, data):
    Hashdata = { id, user, data }
    try:
        tx = await api.transact(
            {
                "actions": [
                    {
                        "account": account,
                        "name": "create",
                        "authorization": [
                            {
                                "actor": actor,
                                "permission": "active"
                            }
                        ],
                        "data": {
                            **Hashdata
                        }
                    }
                ]
            },
            {"broadcast": True, "sign": True}
        )

        print(tx, "\n")
        print("\nResponse data:", tx.processed.action_traces[0].console)
    except Exception as err:
        print(err)

async def DestroyTrancsaction(id):
    try:
        destroyTx = await api.transact(
            {
                "actions": [
                    {
                        "account": account,
                        "name": "destroy",
                        "authorization": [
                            {
                                "actor": actor,
                                "permission": "active"
                            }
                        ],
                        "data": {
                            "id": id
                        }
                    }
                ]
            },
            {"broadcast": True, "sign": True}
        )

        print("Record destroyed ex bgpateng\n\n")
        print(destroyTx, "\n")
        print("responses: \n", destroyTx.processed.action_traces[0].console)
    except Exception as err:
        print(err)

async def PushTransaction(DataId, user, data):
    await CreateTransaction(DataId, user, data)
    await DestroyTrancsaction(DataId)

PushTransaction(1020, account, "tester push transaction ex bgpateng")

File: horangkaya/push/push.mjs
import { Api, JsonRpc, RpcError, JsSignatureProvider } from '../dist/index.js'
import * as dotenv from 'dotenv'
dotenv.config()

const url = process.env.node

const json_rpc = new JsonRpc(url)
const private_key = process.env.pk;

const account = process.env.accinery
const actor = process.env.accinery
const signature  = new JsSignatureProvider([private_key]);

const api = new Api({
    rpc: json_rpc,
    signatureProvider: signature
})

async function create(id, user, data){
    try{
        const tx = await api.transact({
            actions:[
                {
                    account,
                    name:"create",
                    authorization:[
                        {
                            actor,
                            permission:"active"
                        }
                    ],
                    data:{
                        id, user, data
                    }
                }
            ]
        },{broadcast:true,sign:true})
        console.log("{function: create}")
        console.log(tx, "\n")
    }catch(error){
        console.log(error)
    }
}
async function destroy(id){
    try{
        const tx = await api.transact({
            actions:[
                {
                    account,
                    name:"destroy",
                    authorization:[
                        {
                            actor,
                            permission:"active"
                        }
                    ],
                    data:{
                        id
                    }
                }
            ]
        },{broadcast:true,sign:true})
        console.log("{function: destroy}")
        console.log(tx, "\n")
    }catch(error){
        console.log(error)
    }
}

async function main(id, user, data){
    await create(id, user, data)
    await destroy(id)
}

main(3, account, "RPC TEST BY horangkaya")